tfx/experimental/pipeline_testing/dummy_component_launcher.py

```diff
@@ -0,0 +1,102 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""dummy component launcher which launches python executors and dummy executors in process."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import abc
+from typing import Any, Dict, List, Text, Type
+import absl
+
+from tfx import types
+from tfx.components.base import base_executor
+from tfx.orchestration.launcher.in_process_component_launcher import InProcessComponentLauncher
+from tfx.experimental.pipeline_testing.dummy_executor import BaseDummyExecutor
+
+class BaseDummyComponentLauncher(InProcessComponentLauncher):
+  """Responsible for launching a dummy executor.
+
+  The executor will be launched in the same process of the rest of the
+  component, i.e. its driver and publisher.
+  """
+  def __init__(self, **kwargs):
+    super(BaseDummyComponentLauncher, self).__init__(**kwargs)
+    self._record_dir = os.path.join(os.environ['HOME'],
+                                    'tfx/tfx/experimental/pipeline_testing/',
+                                    'testdata')
+    self._executors_dict = {}
+    self._dummy_component_id = []
+    if not os.path.exists(self._record_dir):
+      raise Exception("Must record input/output in {}".format(self._record_dir))
+
+  @abc.abstractmethod
+  def set_dummy_executors(self, component_ids: List[Text],
```

Comment:
Another idea would be a class factory. 
We don't need My....Lancher, with this.

```
def CreateDummyLauncher(record_dir, component_ids, component_map):
  class DummyComponentLauncher(InProcessComponentLauncher):
      def __init__(self, ...):
         super(...).__init__(..)
         self._record_dir = record_dir
         self._dummy_component_ids = component_ids
     ...
  return DummyComponentLauncher
```